tools/toolchains/gcc.py

```diff
@@ -107,6 +111,23 @@ def __init__(self, target,  notify=None, macros=None, build_profile=None,
         self.ar = join(tool_path, "arm-none-eabi-ar")
         self.elf2bin = join(tool_path, "arm-none-eabi-objcopy")
 
+    def version_check(self):
+        stdout, _, retcode = run_cmd([self.cc[0], "--version"], redirect=True)
+        found_version = None
+        for line in stdout.splitlines():
+            for word in line.split():
+                match = self.GCC_VERSION_RE.match(word)
+                if match:
+                    found_version = match.group(0)
+        if found_version and not found_version.startswith(self.GCC_MAJOR + "."):
+            raise NotSupportedException(
```

Comment:
Do we really want to fail the build? Perhaps noticeable warning message will be sufficient?

Have you considered using `LooseVersion` from `distutils.version` package?

Reply:
LooseVersion is a good suggestion. I'll take a look.